### ISSUE TITLE
Sandbox flags should graduate to non-experimental

### ISSUE THREAD

**Issue by AustinSchuh** [bazelbuild/bazel#6345]
@ulfjack mentioned at the Bazel Conference today that we should be filing bugs if there are any flags that we are using which are experimental and required for our builds.

We use the following:

```
--experimental_multi_threaded_digest
--experimental_action_listener
--experimental_sandbox_base=/dev/shm/
--experimental_sandbox_default_allow_network
--experimental_remote_platform_override
--experimental_strict_action_env=true
```

**Comment by dhalperi** [comment]
I don't think this counts by your strict definition, but also building HEAD bazel with bazel requires `--incompatible_remove_native_http_archive=false `

**Comment by dslomov** [comment]
Also cc @buchgr - all these flags are either local or remote execution-related

**Comment by susinmotion** [comment]
@AustinSchuh is this still an open issue?

**Comment by AustinSchuh** [comment]
```
$ cat .bazelrc | grep experimen
build --experimental_multi_threaded_digest
build:coverage --experimental_action_listener=//tools/coverage:copy_source_files_listener
build --experimental_sandbox_base=/dev/shm/
build --experimental_sandbox_default_allow_network=false
build --experimental_strict_action_env=true
```

Looks like we are down 1 flag (Progress!), but the rest are still needed for our build.

**Comment by ulfjack** [comment]
What does your action listener do? Requiring a non-default flag to make coverage work *at all* seems like a bigger issue.

IMO, we should flip the default for experimental_multi_threaded_digest, and make experimental_sandbox_base and experimental_sandbox_default_allow_network non-experimental. I am preparing changes for this right now.

I'm not sure about the experimental_strict_action_env. There's been a bunch of discussion around it, and I think @philwo had a plan, but I don't think anyone worked on it.

**Comment by AustinSchuh** [comment]
Awesome, that's huge progress.  --experimental_strict_action_env still feels experimental from the discussions (6 months ago?) that I last saw.

@ulfjack, I'll give you a better usage of --experimental_action_listener.  I'm a bit fuzzy on what our coverage listener does myself.

We have an extra_action which we use in conjunction with --experimental_action_listener to build up compile_commands.json.  It grabs all the command lines from all the cc_library and cc_binary targets and generates a file used by tools to build.

**Comment by ulfjack** [comment]
Well, --experimental_strict_action_env does not work for everyone, and I believe that the decision was to move forward with a slightly different plan, so that may involve changes to the semantics of the flag.

For --experimental_action_listener, there's been talk about deprecating and removing it in favor of aspects. However, last time I looked at it (which is a while ago), aspects did not have all the features required to replace it.

On a somewhat related note, I'm going to be on vacation for the next 3 weeks starting tomorrow.

**Comment by AustinSchuh** [comment]
3 flags down is huge progress.

We can probably drop --experimental_strict_action_env on our side.  (I've got tools/bazel stripping out the environment as well)
The --experimental_action_listener flag sounds like the harder one.

**Comment by AustinSchuh** [comment]
I just looked, and at least one of these flags, `--experimental_multi_threaded_digest` is still there and we have been using.

**Comment by meisterT** [comment]
That flag is a no-op since https://github.com/bazelbuild/bazel/issues/14034

**Comment by AustinSchuh** [comment]
Digging in again, https://bazel.build/docs/user-manual#flag--experimental_action_listener is the last flag from the original list which is still experimental.

**Comment by meisterT** [comment]
@mai93 I assume `--experimental_action_listener` is deprecated in favor of aspects?

**Comment by mai93** [comment]
that's right, I think aspects have evolved since  https://github.com/bazelbuild/bazel/issues/6345#issuecomment-519803437 but I am not sure what are the current blockers for the migration from `--experimental_action_listener` to aspects

**Comment by meisterT** [comment]
@oquenchil anything left to do here?
